File: lib_common/src/main/java/com/wujia/lib_common/utils/AppContext.java
package com.wujia.lib_common.utils;

import android.content.Context;

public class AppContext {
    public static Context mAppContext;


    public static void init(Context context) {
        if (mAppContext == null) {
            mAppContext = context.getApplicationContext();
        } else {
            throw new IllegalStateException("set context duplicate");
        }
    }

    public static Context get() {
        if (mAppContext == null) {
            throw new IllegalStateException("forget init?");
        } else {
            return mAppContext;
        }
    }
}

File: lib_common/src/main/java/com/wujia/lib_common/utils/AppContext.kt
package com.wujia.lib_common.utils

import android.content.Context

object AppContext {
    var mAppContext: Context? = null


    fun init(context: Context) {
        if (mAppContext == null) {
            mAppContext = context.applicationContext
        } else {
            throw IllegalStateException("set context duplicate")
        }
    }

    fun get(): Context {
        return if (mAppContext == null) {
            throw IllegalStateException("forget init?")
        } else {
            mAppContext!!
        }
    }
}
